kv/dist_sender.go

```diff
@@ -647,42 +671,71 @@ func (ds *DistSender) Send(_ context.Context, call client.Call) {
 			call.Reply.Header().SetGoError(err)
 			return
 		}
+
+		if finalReply != curReply {
+			// This was the second or later call in a multi-range request.
+			// Combine the new response with the existing one.
+			_, ok := finalReply.(proto.Combinable)
+			if !ok {
```

Comment:
That's not equivalent since that code isn't at the end of the surrounding for loop (and even if it were, that seems prone to future errors). I'll stick with the current version.